recipe/0001-conda-Unbundle-libxgboost.-dll-dylib-so.patch

```diff
@@ -1,42 +1,15 @@
-diff --git a/python-package/setup.py b/python-package/setup.py
-index c2e6fd99..ec40ca7c 100644
---- a/python-package/setup.py
-+++ b/python-package/setup.py
-@@ -28,7 +31,12 @@ for libfile in libpath['find_lib_path']():
-         continue
-
- print("Install libxgboost from: %s" % LIB_PATH)
--
-+if os.getenv('CONDA_BUILD'):
-+    print("Conda-build detected, not installing (xgboost nor) libxgboost from: %s" % LIB_PATH)
-+    data_files=('xgboost')
-+else:
-+    print("Install libxgboost from: %s" % LIB_PATH)
-+    data_files=('xgboost', LIB_PATH)
- # Please use setup_pip.py for generating and deploying pip installation
- # detailed instruction in setup_pip.py
- setup(name='xgboost',
-@@ -53,7 +61,7 @@ setup(name='xgboost',
-       # this will use MANIFEST.in during install where we specify additional files,
-       # this is the golden line
-       include_package_data=True,
--      data_files=[('xgboost', LIB_PATH)],
-+      data_files=[data_files],
-       license='Apache-2.0',
-       classifiers=['License :: OSI Approved :: Apache Software License',
-                    'Development Status :: 5 - Production/Stable',
 diff --git a/python-package/xgboost/libpath.py b/python-package/xgboost/libpath.py
-index a313a1b..cd7e1de 100644
+index 65f7403e..4d044cf8 100644
 --- a/python-package/xgboost/libpath.py
 +++ b/python-package/xgboost/libpath.py
-@@ -24,6 +24,10 @@ def find_lib_path():
-                 os.path.join(curr_path, './lib/'),
-                 os.path.join(sys.prefix, 'xgboost')]
-     if sys.platform == 'win32':
+@@ -25,6 +25,10 @@ def find_lib_path():
+         # editable installation, no copying is performed.
+         os.path.join(curr_path, os.path.pardir, os.path.pardir, 'lib'),
```

Comment:
PMFJI, When I had attempted to update for the recipe for a newer version, I too had noticed using `pip` did not work. I had also noticed that the `libxgboost.so` was getting copied into the python package, which I thought was best to avoid. Does this patch here does away with that problem?

Reply:
No, it does not. See https://github.com/dmlc/xgboost/issues/5386.

Reply:
I was assuming that was caused by not using `pip`(my bad!).
Thanks for the pointer.